### ISSUE TITLE
Context.MODE_WORLD_READABLE is deprecated.

### ISSUE THREAD

**Issue by jthrun** [smartdevicelink/sdl_android#311]
The SdlRouter class uses world readable files to announce when it has a transport connection.  The Android docs indicate that this is dangerous and has been deprecated as of API 17.  Further, as of API N (24), use of the `Context.MODE_WORLD_READABLE` mode will throw a SecurityException.

com.smartdevicelink.transport.SdlRouter.java, about line 1427.
```
		@SuppressLint("WorldReadableFiles")
		@SuppressWarnings("deprecation")
		private void storeConnectedStatus(boolean isConnected){
			SharedPreferences prefs = getApplicationContext().getSharedPreferences(getApplicationContext().getPackageName()+SdlBroadcastReceiver.TRANSPORT_GLOBAL_PREFS,
                    Context.MODE_WORLD_READABLE);
            SharedPreferences.Editor editor = prefs.edit();
            editor.putBoolean(SdlBroadcastReceiver.IS_TRANSPORT_CONNECTED, isConnected);
            editor.commit();
		}
```

https://developer.android.com/reference/android/content/Context.html#MODE_WORLD_READABLE

I haven tried it on an N device yet though to see if it actually does throw the exception.

**Comment by joeygrover** [comment]
Yes we are aware. We are investigating solutions and thus why this feature hasn't been merged past develop.

**Comment by jthrun** [comment]
OK, good.  I didn't create a duplicate did I?  I searched and didn't find any similar issues.  We wanted be be able to track this issue.

**Comment by joeygrover** [comment]
No, it wasn't tracked in github yet but was agreed it needed to be fixed before merging to master from other maintainers.

**Comment by joeygrover** [comment]
Resolved with PR #312 merged into develop
